lib/puppet/util/windows/process.rb

```diff
@@ -9,7 +9,13 @@ module Puppet::Util::Windows::Process
   WAIT_TIMEOUT = 0x102
 
   def execute(command, arguments, stdin, stdout, stderr)
-    Process.create( :command_line => command, :startup_info => {:stdin => stdin, :stdout => stdout, :stderr => stderr}, :close_handles => false )
+    arguments[:cwd] ||= Dir.pwd
+
+    # TODO (PUP-6919): Should we wrap a bad/nonexistent CWD in a separate exception? 
+    #
+    # Current syscall error just prints: SystemCallError: The directory name is invalid. - CreateProcess
+    #
+    Process.create( :command_line => command, :startup_info => {:stdin => stdin, :stdout => stdout, :stderr => stderr}, :close_handles => false, :cwd => arguments[:cwd] )
```

Comment:
Seems like it'd be useful to wrap the syscall error to print a more readable error message in case the CWD does not exist or is not a valid PATH format.